diff --git a/libfprint/meson.build b/libfprint/meson.build
index 0ca1767..484b7ab 100644
--- a/libfprint/meson.build
+++ b/libfprint/meson.build
@@ -153,6 +153,8 @@ driver_sources = {
         [ 'drivers/realtek/realtek.c' ],
     'focaltech_moc' :
         [ 'drivers/focaltech_moc/focaltech_moc.c' ],
+    'vfs0090' :
+        [ 'drivers/vfs0090/vfs0090.c' ],
 }
 
 helper_sources = {
@@ -166,6 +168,8 @@ helper_sources = {
         [ ],
     'udev' :
         [ ],
+    'nss' :
+        [ ],
     'virtual' :
         [ 'drivers/virtual-device-listener.c' ],
 }
diff --git a/meson.build b/meson.build
index baafa19..9fc64e1 100644
--- a/meson.build
+++ b/meson.build
@@ -178,6 +178,7 @@ endian_independent_drivers = virtual_drivers + [
     'vcom5s',
     'vfs101',
     'vfs7552',
+    'vfs0090',
 ]
 
 all_drivers = default_drivers + virtual_drivers
@@ -211,6 +212,7 @@ driver_helper_mapping = {
     'aes4000' : [ 'aeslib', 'aes3k' ],
     'uru4000' : [ 'openssl' ],
     'elanspi' : [ 'udev' ],
+    'vfs0090' : [ 'aes3k', 'openssl', 'nss' ],
     'virtual_image'          : [ 'virtual' ],
     'virtual_device'         : [ 'virtual' ],
     'virtual_device_storage' : [ 'virtual' ],
@@ -273,6 +275,13 @@ foreach i : driver_helpers
         endif
 
         optional_deps += openssl_dep
+    elif i == 'nss'
+        nss_dep = dependency('nss', required: false)
+        if not nss_dep.found()
+            error('NSS is required for @0@ and possibly others'.format(driver))
+        endif
+
+        optional_deps += nss_dep
     elif i == 'udev'
         install_udev_rules = true
 
